requestvalidators.go

```diff
@@ -0,0 +1,74 @@
+package pinecone
+
+import "fmt"
+
+// validateQueryParams validates the query parameters.
+func validateQueryParams(params QueryParams) error {
+	if params.TopK < 1 {
+		return fmt.Errorf("%w: top k is required and must be greater than 0", ErrInvalidParams)
+	}
+
+	if params.Vector == nil && params.Id == "" {
+		return fmt.Errorf("%w: vector or id is required", ErrInvalidParams)
+	}
+
+	if params.Vector != nil && params.Id != "" {
+		return fmt.Errorf("%w: cannot specify both vector and id", ErrInvalidParams)
+	}
+
+	if params.SparseVector != nil && len(params.SparseVector.Values) != len(params.SparseVector.Indices) {
+		return fmt.Errorf("%w: sparse vector values and indices must be the same length", ErrInvalidParams)
+	}
+
+	return nil
+}
+
+// validateDeleteVectorsParams validates the delete vectors parameters.
+func validateDeleteVectorsParams(params DeleteVectorsParams) error {
+	if params.Ids == nil && len(params.Ids) < 1 && !params.DeleteAll {
```

Comment:
it is safe to use `len()` to obtain and determine the length of the slice without `== nil` expression